src/main/scala/org/ergoplatform/modifiers/history/BlockTransactions.scala

```diff
@@ -5,24 +5,55 @@ import io.circe.{Decoder, Encoder, HCursor}
 import org.ergoplatform.http.api.ApiCodecs
 import org.ergoplatform.modifiers.BlockSection
 import org.ergoplatform.modifiers.mempool.{ErgoTransaction, ErgoTransactionSerializer}
+import org.ergoplatform.nodeView.mempool.TransactionMembershipProof
 import org.ergoplatform.settings.{Algos, Constants}
 import scorex.core._
 import scorex.core.serialization.ScorexSerializer
 import scorex.crypto.authds.LeafData
+import scorex.crypto.authds.merkle.{Leaf, MerkleProof, MerkleTree}
 import scorex.crypto.hash.Digest32
 import scorex.util.serialization.{Reader, Writer}
 import scorex.util.{ModifierId, bytesToId, idToBytes}
 import scorex.util.Extensions._
 
-case class BlockTransactions(headerId: ModifierId, txs: Seq[ErgoTransaction], override val sizeOpt: Option[Int] = None)
-  extends BlockSection
-    with TransactionsCarryingPersistentNodeViewModifier[ErgoTransaction] {
+
+/**
+  * Section of a block which contains transactions.
+  *
+  * @param headerId - identifier of a header of a corresponding block
+  * @param txs - transactions of a block
+  * @param sizeOpt - (optional) size of the section (cached to not be calculated again)
+  */
+case class BlockTransactions(headerId: ModifierId,
+                             txs: Seq[ErgoTransaction],
+                             override val sizeOpt: Option[Int] = None)
+  extends BlockSection with TransactionsCarryingPersistentNodeViewModifier[ErgoTransaction] {
 
   assert(txs.nonEmpty, "Block should always contain at least 1 coinbase-like transaction")
 
   override val modifierTypeId: ModifierTypeId = BlockTransactions.modifierTypeId
 
-  override def digest: Digest32 = BlockTransactions.transactionsRoot(txs)
+  lazy val txIds: Seq[Array[Byte]] = txs.map(_.serializedId)
```

Comment:
Opportunity for `NonEmptyList`?

Reply:
@greenhat from which framework ?

Reply:
I believe we should have `cats` in dependencies. So  (`NonEmptyList`)[https://typelevel.org/cats/datatypes/nel.html] seems like a good fit.